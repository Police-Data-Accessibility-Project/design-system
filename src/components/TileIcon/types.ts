export interface PdapTileIconProps {
	imgAltText: string;
	imgSrc: string;
}
