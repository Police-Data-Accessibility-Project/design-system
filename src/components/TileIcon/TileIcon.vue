<template>
	<img class="pdap-tile-icon" :alt="imgAltText" :src="imgSrc" />
</template>

<script setup lang="ts">
import { PdapTileIconProps } from './types';

defineProps<PdapTileIconProps>();
</script>

<script lang="ts">
/**
 * @deprecated unused
 */
export default {
	name: 'TileIcon',
};
</script>

<style>
.pdap-tile-icon {
	@apply h-auto max-w-[7.5rem];
}
</style>
