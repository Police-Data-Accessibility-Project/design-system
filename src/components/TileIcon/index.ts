import TileIcon from './TileIcon.vue';

export { TileIcon };
