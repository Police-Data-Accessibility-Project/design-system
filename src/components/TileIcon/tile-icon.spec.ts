// Component
import TileIcon from './TileIcon.vue';

// Utils
import { mount } from '@vue/test-utils';
import { describe, expect, test } from 'vitest';

// Test
describe('Tile icon component', () => {
	// Render
	test('Renders a Tile icon', () => {
		const wrapper = mount(TileIcon, {
			props: {
				imgSrc: 'foo',
				imgAltText: 'bar',
			},
		});

		expect(wrapper.find('.pdap-tile-icon').exists()).toBe(true);
		expect(wrapper.props()).toStrictEqual({ imgSrc: 'foo', imgAltText: 'bar' });
		expect(wrapper.html()).toMatchSnapshot();
	});
});
