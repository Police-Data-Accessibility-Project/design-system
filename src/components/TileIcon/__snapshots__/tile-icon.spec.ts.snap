// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Tile icon component > Renders a Tile icon 1`] = `<img alt="bar" class="pdap-tile-icon" src="foo">`;
